### DISCUSSION TITLE
mpcsetup v0.10

### DISCUSSION THREAD

**Discussion by 0xForerunner** [Consensys/gnark#1129] [General]
I see that there is the mpcsetup template in this repo with a bunch of utility functions. Is there a premade cli anywhere for the setup or is the best route to make my own? Would be really cool if these templates also generated the cli for this.

**Comment by gbotrel** [comment]
Hi, you can check out Binance  CLI https://github.com/bnb-chain/zkbnb-setup (under Apache 2.0).

**Reply by hussein-aitlahcen** [reply]
I see that they use the trusted setup for the pedersen key: https://github.com/bnb-chain/zkbnb-setup/blob/ca7f469d477cdf5719081d4b76dd8cf9830e4d04/keys/keys.go#L544

Is this considered sound if the rest has been MPC generated?

**Reply by ivokub** [reply]
My initial intuition is that it is sound - the random G2 factor is used for hiding property, but we're relying only on the binding property. But in that case it would make sense to fix G2 point instead of generating new.

cc @Tabaie.

**Reply by Tabaie** [reply]
I think the Pedersen key also needs to be MPC generated, if our threat model involves not trusting any single party with knowing the setup parameters
 If a sole malicious party generates/knows the $\sigma$ factor they can pass off any $h\in G_1$ as a legitimate commitment by producing $\pi = h^\sigma$, thus breaking soundness.

I agree that $g\in G_2$ can be fixed.

**Reply by HSG88** [reply]
As @Tabaie mentioned, Pedersen key needs to be generated as part of the MPC setup. The use case in zkbnb didn't involve using Pedersen key and so it wasn't part of the MPC generated parameters.

**Reply by 0xForerunner** [reply]
@gbotrel Thanks for the link. We were previously using a fork of that, but it's still on gnark 0.8 and there've been quite a few changes since then. I've started building a new version from scratch. @HSG88 or @Tabaie Could you elaborate on what the process should be for properly generating the Pederson key? I'm currently using `mpcsetup.ExtractKeys` to generate the vk, but that function leaves the pederson key empty.

Also, if you're interested, [this](https://github.com/worldcoin/semaphore-mtb-setup/tree/ewoolsey/gnark-10) is the branch I'm working on for the new setup.

**Reply by 0xForerunner** [reply]
@Tabaie @HSG88 I've started some work generating the pedersen key via MPC. Made a post [here](https://github.com/Consensys/gnark/discussions/1172). Looking for some feedback/assistance.
